src/test/java/edu/hm/hafner/util/SetTest.java

```diff
@@ -0,0 +1,180 @@
+package edu.hm.hafner.util;
+
+import java.util.Collections;
+import java.util.HashSet;
+import java.util.Iterator;
+
+import org.assertj.core.api.SoftAssertions;
+import org.junit.jupiter.api.Test;
+
+import static org.assertj.core.api.Assertions.*;
+
+/**
+ * Tests the class {@link HashSet}.
+ *
+ * @author Dominik Moelter
+ */
+public class SetTest {
+
+    HashSet createEmptyHashSet() {
+        return new HashSet<Integer>();
+    }
+
+    HashSet createFilledHashSet(final Integer... values) {
+        HashSet<Integer> hashSet = new HashSet<Integer>();
+        Collections.addAll(hashSet, values);
+        return hashSet;
+    }
+
+    @Test
+    void testConstructorCapacity() {
+        assertThatThrownBy(() -> new HashSet<Integer>(-1))
+                .as("Negative capacity throws IllegalArgumentException")
+                .isExactlyInstanceOf(IllegalArgumentException.class);
+    }
+
+    @Test
+    void testConstructorCapacityLoadFactor() {
+        SoftAssertions softAssertions = new SoftAssertions();
+
+        softAssertions.assertThatThrownBy(() -> new HashSet<Integer>(-16, (float) 0.75))
+                .as("Negative capacity throws IllegalArgumentException")
+                .isExactlyInstanceOf(IllegalArgumentException.class);
+        softAssertions.assertThatThrownBy(() -> new HashSet<Integer>(16, (float) -0.75))
+                .as("Negative capacity throws IllegalArgumentException")
+                .isExactlyInstanceOf(IllegalArgumentException.class);
+
+        softAssertions.assertAll();
+    }
+
+    @Test
+    void testConstructorCollection() {
```

Comment:
Um diesen Konstruktor zu testen wäre auch ein Test mit einer gefüllten Collection als Konstruktor-Parameter sinnvoll.

Reply:
Ich konnte leider online keine Informationen finden wie ich testen kann, dass keine Exception geworfen wird. Durch die Code completion konnte ich die  zweite Assertion schreiben. Leider funktioniert diese nicht. Der Test schlägt mit der Meldung 'Expecting code to raise a throwable.' fehl.

`    @Test
    void testConstructorCollection() {
        SoftAssertions softAssertions = new SoftAssertions();
        ArrayList<Integer> arrayList = new ArrayList<>(Arrays.asList(1, 2, 3, 4));

        softAssertions.assertThatThrownBy(() -> new HashSet<Integer>(null))
                .as("Null collection throws NullPointerException")
                .isExactlyInstanceOf(NullPointerException.class);
        softAssertions.assertThatThrownBy(() -> new HashSet<Integer>(arrayList))
                .doesNotThrowAnyException();

        softAssertions.assertAll();
    }
`

Wie setzt man einen solche Assertion in um?

Reply:
So z.B.:
`softAssertions.assertThatCode(() -> new HashSet<Integer>(arrayList))
                .doesNotThrowAnyException();`
Alternativ führt man den Code einfach aus und überprüft ob die Elemente im neuen HashSet enthalten sind. Wenn eine Exception fliegen würde, würde der Test automatisch als fehlgeschlagen Zählen.